scripts/compile-references.js

```diff
@@ -25,25 +25,20 @@
  * only when required, but it cannot infer workspaces by itself, it has to be
  * explicitly defined [1].
  *
- * You can do a dry run using the cli flag `--dry-run`, the script will exit
- * with a code different from zero if something needed to be updated.
- *
  * [1]: https://www.typescriptlang.org/docs/handbook/project-references.html
  */
 
 
 const cp = require('child_process');
-const path = require('path').posix;
+const path = require('path');
 const fs = require('fs');
 
 const ROOT = path.join(__dirname, '..');
 
-const DRY_RUN = popFlag(process.argv, '--dry-run');
-
-const FORCE_REWRITE = popFlag(process.argv, '--force-rewrite') && !DRY_RUN;
+const FORCE_REWRITE = process.argv.includes('--force-rewrite');
 
 /** @type {{ [packageName: string]: YarnWorkspace }} */
-const YARN_WORKSPACES = JSON.parse(cp.execSync('yarn --silent workspaces info').toString());
+const YARN_WORKSPACES = JSON.parse(cp.execSync('yarn --silent workspaces info', { cwd: ROOT }).toString());
```

Comment:
I don't quite understand why we need to rewrite references as part of the build at all: aren't they static and inferred by the usage relation between packages?

Reply:
They are static and inferred. The reason we do it on build is to make sure everything is in sync since building relies on it.

Reply:
Note that it won't write anything on disk if everything is already synced.

Reply:
So I _did_ understand that right. It's a pity the tsc compiler can't infer those if we tell it what the set of source packages are 🤷  Fair enough.